### PR TITLE
Fix express.static() with esModuleInterop

### PR DESCRIPTION
**Pull request by tkrotoff**
TLDR; `esModuleInterop` (TypeScript 2.7) does not like definitions with a function and a namespace having the same name.
<br><br>

~~/!\ Not to be merged, please discuss~~
<br><br>

Example with [serve-static](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/a436740d5ddaff62e522da2ae212031acb7ef935/types/serve-static) (this problem will happen with other definitions).

- [serve-static/index.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a436740d5ddaff62e522da2ae212031acb7ef935/types/serve-static/index.d.ts):
```TypeScript
declare function serveStatic(...): ...;
declare namespace serveStatic {
  ...
}
export = serveStatic;
```

- [express/index.d.ts](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a436740d5ddaff62e522da2ae212031acb7ef935/types/express/index.d.ts):
```TypeScript
import * as serveStatic from 'serve-static';
declare namespace e {
  var static: typeof serveStatic;
  ...
}
export = e;
```

- [Use of express.static()](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/a436740d5ddaff62e522da2ae212031acb7ef935/types/express/express-tests.ts):
```TypeScript
import express from 'express';
const server = express();
express.static.mime.define({'application/fx': ['fx']});
server.use(express.static('.')); // With esModuleInterop: Cannot invoke an expression whose type lacks a call signature. Type 'typeof serveStatic' has no compatible call signatures.
```

Error with `esModuleInterop`: `Cannot invoke an expression whose type lacks a call signature. Type 'typeof serveStatic' has no compatible call signatures.`

If inside express/index.d.ts, `import * as serveStatic` is changed to `import serveStatic` then it works with `esModuleInterop`. Problem: it does not work without it (`Module '"serve-static/index"' has no default export`).

Any suggestion?

### CODE REVIEW

**Comment by tkrotoff** [comment]
Ping TypeScript gurus @andy-ms @sheetalkamat

**Comment by andy-ms** [comment]
That should probably be `import serveStatic = require("serve-static")` in `express/index.d.ts` -- could you see if that works for you?

**Comment by tkrotoff** [comment]
Thx @andy-ms 
`require("serve-static")` works with and without `esModuleInterop` :-)

I've rebased the PR, this can be merged

File: types/express/index.d.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@
 /// <reference types="serve-static" />
 
 import * as bodyParser from "body-parser";
-import * as serveStatic from "serve-static";
+import serveStatic = require("serve-static"); // FIXME See Fix express.static() with esModuleInterop https://github.com/DefinitelyTyped/DefinitelyTyped/pull/23335
```
Comment (andy-ms) [inline types/express/index.d.ts]: I don't think the comment is needed if this fixes the issue?

Comment (tkrotoff) [reply types/express/index.d.ts]: If someone passes by and finds it strange (*) to have `require` instead of `from`, he has the link to the explanation

(*) and wants to revert the change

Comment (andy-ms) [reply types/express/index.d.ts]: It's actually more correct this way -- the thing being imported is a function. See the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#should-i-add-an-empty-namespace-to-a-package-that-doesnt-export-a-module-to-use-es6-style-imports).

Comment (tkrotoff) [reply types/express/index.d.ts]: @andy-ms I've removed the comment
